macros/src/lib.rs

```diff
@@ -23,14 +23,23 @@ pub fn statemachine(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
         Ok(sm) => {
             #[cfg(feature = "graphviz")]
             {
+                use std::hash::{Hash, Hasher};
                 use std::io::Write;
 
+
                 // Generate dot syntax for the statemachine.
                 let diagram = diagramgen::generate_diagram(&sm);
+                let diagram_name = if let Some(name) = &sm.name{
+                    name.to_string()
+                }else{
+                    let mut diagram_hasher = std::collections::hash_map::DefaultHasher::new();
+                    diagram.hash(&mut diagram_hasher);
+                    format!("{:010x}", diagram_hasher.finish())
+                };
 
                 // Start the 'dot' process.
                 let mut process = std::process::Command::new("dot")
-                    .args(&["-Tsvg", "-o", "statemachine.svg"])
+                    .args(&["-Tsvg", "-o", &format!("statemachine_{diagram_name}.svg")])
```

Comment:
Hmm, this means that if people don't name their state machine, they get a file like `statemachine_<HASHDATA>.svg`, which seems fine? Maybe we could add an `smlang` label to it to, but don't feel pressed on this